lib/task_005_lib.rb

```diff
@@ -0,0 +1,95 @@
+module Task005
+  module_function
+
+  class Point
+    attr_reader :x, :y
+
+    def self.from_input(point)
+      x, y = point.split(',').map(&:to_i)
+      Point.new(x, y)
+    end
+
+    def initialize(x, y)
+      @x = x
+      @y = y
+    end
+  end
+
+  class Line
+    attr_reader :point1, :point2
+
+    def self.from_input(input)
+      point1, point2 = input.split(' -> ').map { |point| Point.from_input(point) }
+      Line.new(point1, point2)
+    end
+
+    def initialize(point1, point2)
+      @point1 = point1
+      @point2 = point2
+    end
+
+    def horizontal_or_vertical?
+      @point1.x == @point2.x || @point1.y == @point2.y
+    end
```

Comment:
What do you think about splitting it even further? :)

```suggestion
    def horizontal_or_vertical?
      horizontal? || vertical?
    end
    
    def horizontal?
      @point1.y == @point2.y
    end
    
    def vertical?
      @point1.x == @point2.x
    end
```

To me, in this case, it would be easier to comprehend.